### DISCUSSION TITLE
I don't know either

### DISCUSSION THREAD

**Discussion by carltongibson** [carltongibson/notes#33] [Being an open source maintainer]
I posted some thoughts on normalising saying "I don't know". I think new or prospective maintainers can be helped a lot by showing that you don't always (often don't!) know what the right approach forward is...

https://noumenal.es/notes/open-source/I-dont-know-either/

What do you think?

**Comment by nanorepublica** [comment]
The phrase just makes me think of parenting advice 🙈 admitting to your kids (new contributors?) that you don't have all the answers. 

So yes I agree on using it, I then typically add on "What do you think?" - With children this can interesting tangental answers, but I would hope a fellow adult might answer more logically?

**Reply by bckohan** [reply]
I think all aspects of society could be improved by more acknowledgement of uncertainty where it exists.

My wife and I just welcomed our first two weeks ago and I'm already amazed by how many professionals and sub-professionals in the child care space will tell you completely contradictory things with confidence.  I'm desperate for people to just kind of shrug at my anxieties and say things like "Whatever you decide is probably fine".

A corollary to "I don't know" may be "Conventional wisdom isn't always correct.". I had a use case a couple years ago where I had to query many related boolean flags on a large database table. I found lots of advice saying it wasn't worth the trouble to squish these boolean columns into a bit mask. I thought - could this possibly be true? [Turns out it isn't.](https://django-enum.readthedocs.io/en/latest/performance.html#flags)

**Reply by carltongibson** [reply]
Congratulations @bckohan 🥳 enjoy the journey. 

The enum case there is a nice one. I'll have a play with that. 

I think in general I'd recommend "presume the existing decision right"
as a better starting point though. **sometimes** it's not, but more times than not (by a vast margin) it'll lead you better. 

It's related to the "is the bug in my code or the library?" question. Almost always in your own code. (But sometimes not.)

**Reply by bckohan** [reply]
@carltongibson Thanks! She's already the best thing I've ever done.
